src/components/partnership/partnership.service.ts

```diff
@@ -369,6 +384,10 @@ export class PartnershipService {
       }
     }
 
+    if (input.primary) {
```

Comment:
We should check that the previous value is not primary so we don't call move unnecessarily.